src/scenes/Projects/List/ProjectList.tsx

```diff
@@ -40,13 +44,46 @@ export const ProjectList: FC = () => {
       },
     },
   });
+
+  useEffect(() => {
+    switch (activeTab) {
+      case 'all':
+      case 'pinned':
+        setFilters({ ...filters, mine: false });
+        break;
+      case 'mine':
+        setFilters({ ...filters, mine: true });
+        break;
+      default:
+        break;
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [activeTab]);
+
+  const pinnedProjects = useMemo(
+    () => data?.projects.items.filter((project) => project.pinned),
+    [data?.projects.items]
+  );
+
+  const getProjectTotal = () => {
+    switch (activeTab) {
+      case 'mine':
+      case 'all':
+        return formatNumber(data?.projects.total);
+      case 'pinned':
+        return formatNumber(pinnedProjects?.length);
+      default:
+        return 0;
+    }
+  };
```

Comment:
`pinned` needs to be sent in the API filters, so they API can give us the pinned ones and allow us to pagination through them. With that filter the total will be correct so it can come straight from the API result.

Reply:
If we pin a project in the `All` tab, it should be added immediately in the `Pinned` tab. Since we don't re-fetch API after toggling the project,  we cannot get the newly added pinned project. So I filtered pinned projects from all projects list like below.
```
const pinnedProjects = useMemo(
    () => data?.projects.items.filter((project) => project.pinned),
    [data?.projects.items]
);
```
FYI, we can get the correct pinned projects and the total number if we add the `pinned` filter in the API filters.

Reply:
Filtering client side won't work because the client doesn't have all the information. A user could have a pinned project that's on page 5. Sourcing the pinned list from the all list wouldn't grab that project if only the first 4 pages are loaded.

I've got a branch to help add items to lists in cache. Let's just ignore this problem until that branch lands.

Reply:
Thanks